service/sharddistributor/handler/handler.go

```diff
@@ -34,13 +35,17 @@ import (
 	"github.com/uber/cadence/service/sharddistributor/constants"
 )
 
+const (
+	_heartbeatRefreshRate = 2 * time.Second
+)
+
 func NewHandler(
 	logger log.Logger,
 	metricsClient metrics.Client,
 	matchingRing membership.SingleProvider,
 	historyRing membership.SingleProvider,
-) Handler {
-	handler := &handlerImpl{
+) *Impl {
```

Comment:
Here we return a struct ptr, but in the executor we return the interface, we should probably be consistent.